File: src/MindBloom.Application/Features/Auth/DTOs/RegisterRequest.cs
namespace MindBloom.Application.Features.Auth.DTOs;

public class RegisterRequest
{
    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Username { get; set; } = string.Empty;

    public string Password { get; set; } = string.Empty;

    public DateTime DateOfBirth { get; set; }
    public string Role { get; set; } = null!;
}

File: MindBloom.Frontend/mindbloom_mobile/lib/features/auth/data/models/register_request.dart
class RegisterRequest {
  final String firstName;
  final String lastName;
  final String username;
  final String email;
  final String password;
  final DateTime dateOfBirth;

  RegisterRequest({
    required this.firstName,
    required this.lastName,
    required this.username,
    required this.email,
    required this.password,
    required this.dateOfBirth,
  });

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'username': username,
      'email': email,
      'password': password,
      'dateOfBirth': dateOfBirth.toIso8601String(),
      'role': 'Client',
    };
  }
}
